client/my-sites/promote-post-i2/components/ad-preview/index.tsx

```diff
@@ -1,31 +1,58 @@
-import { useEffect, useRef } from 'react';
 import './style.scss';
+import { useEffect } from 'react';
 
 interface Props {
 	htmlCode: string;
+	templateFormat: string;
 	isLoading?: boolean;
 }
 
-export default function AdPreview( { htmlCode, isLoading }: Props ) {
-	const shadowHost = useRef< HTMLDivElement >( null );
-
+export default function AdPreview( { htmlCode, isLoading, templateFormat }: Props ) {
 	useEffect( () => {
-		if ( ! isLoading && htmlCode && shadowHost.current ) {
-			const shadowRoot = shadowHost.current.attachShadow( { mode: 'open' } );
-			shadowRoot.innerHTML = `
-          <div class="viewer viewer__receipt}">
-            ${ htmlCode }
-          </div>`;
+		if ( ! isLoading && templateFormat === 'html5_v2' ) {
+			// we only need this listener to resize the iframe for html5_v2 templates
+			window.addEventListener( 'message', function ( msg ) {
+				if ( typeof msg.data !== 'object' ) {
+					return;
+				}
+
+				if ( msg.data.type !== 'wa-inline-frame' ) {
+					return;
+				}
+
+				const iframes = document.getElementsByTagName( 'iframe' );
+
+				for ( let i = 0; i < iframes.length; i++ ) {
```

Comment:
we wont have this problem because we check and pick the specific iframe 
this happens in this line 

`if ( iframes[ i ].contentWindow === msg.source ) {`

this will check that the msg.source comes from the iframe that we are resizing. 

even though in calypso and on this page we will never have more than 1 iframe, it is possible that in a real page delivering ads we might have more than 1. I tested this thoroughly, and made sure that all iframes resize correctly